### PR TITLE
refactored to work on Android N

### PR DESCRIPTION
**Pull request by sudochownrwhoami**

### CODE REVIEW

**Comment by sudochownrwhoami** [comment]
Hi facebook-github-bot,

I signed up. That aside:

I'm attempting to provide a working version of the Parse Android SDK specifically for the Android N Developer Preview

The automated build failed because of the gradle version I am using that is apparently required for the Android N dev preview builds

**Comment by grantland** [comment]
It's awesome that you've looked into this! Is there a problem using the SDK on Android N or is this allowing the SDK to compile with Android N?

We currently have an issue for getting the SDK to compile under Android M as it relies on APIs removed in M, but it works on M without any changes: #189

It seems like you've done the basic work outlined in that issue, however, your work will also break compatibility with < android-11 since `Notification.Builder` requires android-11 and I don't think we're ready to remove support for < android-11.

Additionally, you can update `.travis.yml` to get the tests passing.

**Comment by grantland** [comment]
Still waiting for working Travis-CI config

File: Parse/build.gradle [hunk 1]
```diff
@@ -36,14 +38,22 @@ android {
...
+
+
+    useLibrary 'org.apache.http.legacy'
 }
 
 dependencies {
+    compile 'com.android.tools.build:gradle:2.0.0-beta6'
     compile 'com.parse.bolts:bolts-tasks:1.4.0'
-
+    compile 'org.apache.httpcomponents:httpclient-android:4.3.5.1'
```
Comment (Jawnnypoo) [inline Parse/build.gradle]: This is unnecessary since you already have `useLibrary 'org.apache.http.legacy'`

**Comment by Jawnnypoo** [comment]
This should be updated to use non-beta versions of things like the compile SDK, build tools, etc.

File: Parse/build.gradle [hunk 2]
```diff
@@ -9,20 +9,22 @@ version = '1.13.1-SNAPSHOT'
 buildscript {
     repositories {
         mavenCentral()
```
Comment (rogerhu) [inline Parse/build.gradle]: you can kill this line, maven is redundant

File: Parse/build.gradle (hunk 1 above)
Comment (rogerhu) [inline Parse/build.gradle]: delete line

File: Parse/build.gradle [hunk 3]
```diff
@@ -204,6 +214,6 @@ task jacocoTestReport(type:JacocoReport, dependsOn: "testDebugUnitTest") {
 
 //region Coveralls
 
-coveralls.jacocoReportPath = "${buildDir}/reports/jacoco/jacocoTestReport/jacocoTestReport.xml"
```
Comment (rogerhu) [inline Parse/build.gradle]: can you add this back in?

File: Parse/build.gradle [hunk 4]
```diff
@@ -36,14 +38,22 @@ android {
...
+
+    useLibrary 'org.apache.http.legacy'
 }
 
 dependencies {
+    compile 'com.android.tools.build:gradle:2.0.0-beta6'
     compile 'com.parse.bolts:bolts-tasks:1.4.0'
-
+    compile 'org.apache.httpcomponents:httpclient-android:4.3.5.1'
     provided 'com.squareup.okhttp:okhttp:2.4.0'
```
Comment (rogerhu) [inline Parse/build.gradle]: these need to go to okhttp3

**Comment by Jawnnypoo** [comment]
@rogerhu Would you be willing to accept https://github.com/ParsePlatform/Parse-SDK-Android/pull/554 instead? Reasoning: all of the work here is included in that PR and is ready to merge, and reducing down to just the OkHttp client reduces the amount of support needed for the project.

**Comment by rogerhu** [comment]
I think we need to get OkHttp3 stabilized with some of the more minor issues (i.e. v1.13.2 or v1.13.3) first -- waiting to hear back about the process to push to Sonatype/Maven, or if it's already automated if I push/tag a release. 

But let me look at that other PR..

**Comment by rogerhu** [comment]
Closing in lieu of https://github.com/ParsePlatform/Parse-SDK-Android/pull/554

**Comment by Jawnnypoo** [comment]
Sounds good. If nothing else, the library could be moved to a JitPack distribution since that requires no authentication to distribute so long as you can make tags in GitHub.

**Pull request closed without merging**